From 0724e7da5bec7eba7d209c65806e335ec142e1e0 Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <skardach@marvell.com>
Date: Fri, 8 Feb 2019 11:05:01 +0100
Subject: [PATCH 0966/1051] soc: octeontx2-rm: enable bus master

Bus mastering for RVU TIM/SSO device is required for MSIX interrupt
reception.

Change-Id: Id24212d55a0375fe73565f221eef0f6bdb9f733b
Signed-off-by: Stanislaw Kardach <skardach@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3700
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-rm/otx2_rm.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
index 24838ab7c7c3..293bc343b9a7 100644
--- a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
+++ b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
@@ -1154,6 +1154,8 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		goto set_mask_failed;
 	}
 
+	pci_set_master(pdev);
+
 	/* CSR Space mapping */
 	rm->bar2 = pcim_iomap(pdev, PCI_CFG_REG_BAR_NUM,
 			       pci_resource_len(pdev, PCI_CFG_REG_BAR_NUM));
@@ -1165,7 +1167,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	err = rm_check_pf_usable(rm);
 	if (err)
-		goto set_mask_failed;
+		goto pf_unusable;
 
 	/* Map PF-AF mailbox memory */
 	rm->af_mbx_base = ioremap_wc(pci_resource_start(pdev, PCI_MBOX_BAR_NUM),
@@ -1173,7 +1175,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (!rm->af_mbx_base) {
 		dev_err(&pdev->dev, "Unable to map BAR4\n");
 		err = -ENODEV;
-		goto afmbox_mapfailed;
+		goto pf_unusable;
 	}
 
 	/* Request IRQ for PF-VF mailbox here - TBD: check if this can be moved
@@ -1225,7 +1227,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	rm_free_irqs(pdev);
 alloc_irqs_failed:
 	iounmap(rm->af_mbx_base);
-afmbox_mapfailed:
+pf_unusable:
 	pcim_iounmap(pdev, rm->bar2);
 set_mask_failed:
 	pci_release_regions(pdev);
-- 
2.17.1

